src/CortexMock.cpp

```diff
@@ -0,0 +1,855 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <thread>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "ros2_cortex/CortexMock.hpp"
+#include "rapidjson/filereadstream.h"
+#include "rapidjson/writer.h"
+#include "rapidjson/stringbuffer.h"
+
+namespace ros2_cortex
+{
+
+CortexMock::CortexMock(std::string & capture_file_name)
+: capture_file_name_(capture_file_name)
+{
+  initReadFile();
+}
+
+void CortexMock::initReadFile()
+{
+  FILE * fp = fopen(capture_file_name_.data(), "r");
+
+  std::vector<char> read_buffer(read_buffer_size_);
+  rapidjson::FileReadStream is(fp, read_buffer.data(), read_buffer_size_);
+
+  document_.ParseStream(is);
+  fclose(fp);
+  n_frames_ = document_["framesArray"].Size();
+  // TODO(Gergely Kovacs) capture bodydefs too for the json files
+  // extractBodyDefs(body_defs_, document_["bodyDefs"]);
+}
+
+CortexMock::CortexMock(const CortexMock & other)
+: verbosity_level_(other.verbosity_level_), analog_bit_depth_(other.analog_bit_depth_),
+  conv_rate_to_mm_(other.conv_rate_to_mm_), frame_rate_(other.frame_rate_), analog_sample_rate_(
+    other.analog_sample_rate_),
+  axis_up_(other.axis_up_), dataHandlerFunc_(other.dataHandlerFunc_), errorMsgHandlerFunc_(
+    other.errorMsgHandlerFunc_), capture_file_name_(other.capture_file_name_),
+  current_framenum_(other.current_framenum_)
+{
+  // the result of the copy-constructor isn't active yet, even if the other one was active
+  // TODO(Gergely Kovacs) if we connect to client, addresses and ports need to be copied, too
+  initReadFile();
+}
+
+void CortexMock::swap(CortexMock & other) const noexcept
+{
+  // E.g. if pointers need to be swapped
+}
+
+CortexMock & CortexMock::operator=(CortexMock other)
+{
+  other.swap(*this);  // Copy-constructor and non-throwing swap
+  return *this;
+}
+
+CortexMock::~CortexMock()
+{
+  run_thread.join();
+  freeFrame(&current_frame_);
+}
+
+void CortexMock::getCaptureFilename(std::string & dest) const
+{
+  dest = capture_file_name_;
+}
+
+int CortexMock::getSdkVersion(unsigned char version[4]) const
+{
+  version[0] = 0;
+  version[1] = 8;
+  version[2] = 0;
+  version[3] = 0;
+  return RC_Okay;
+}
+
+int CortexMock::setVerbosityLevel(int i_level)
+{
+  verbosity_level_ = i_level;
+  return RC_Okay;
+}
+
+int CortexMock::getVerbosityLevel() const
+{
+  return verbosity_level_;
+}
+
+void CortexMock::errorMsgInString(int i_level, std::string & msg_str) const
+{
+  if (verbosity_level_ >= i_level) {
+    errorMsgHandlerFunc_(i_level, &msg_str[0]);
+  }

```

Comment:
Cortex has a verbosity level itself which can be set and thus the mock too. And errorMsgHandlerFunc_ can't know this level, because it only has 2 parameters which are used for something else. So I don't think this should be changed. Do you still?

Reply:
What does the verbosity level of Cortex configure? Does it change whether certain messages are treated as INFO/WARN/ERROR, or does it control whether messages labeled as INFO/DEBUG etc. should be included in the log/console? It is also strange for me that an INFO message would be handled by a function called errorMsgHandlerFunc_

Reply:
It says it's the "filter level of the LogMessages", so the latter. It's an unfortunate name, but I think this is the way it was intended. But take a look at functions get/setVerbosityLevel and setErrorMsgHandler of Cortex.h.